src/reassign.c

```diff
@@ -1,19 +1,62 @@
 #define USE_RINTERNALS
 #include <R.h>
+#include <Rinternals.h>
 #include <R_ext/Error.h>
 #include <R_ext/Rdynload.h>
 #include <Rdefines.h>
 #include <stdlib.h>  // for NULL
+#include <stdint.h> // for uint64_t
+
+// Mirror the exact structures of SEXPREC from R internals
+struct proxy_sxpinfo_struct {
```

Comment:
I'd move the comments here just because this is the first thing that you read where you wonder what's going on.

Reply:
Move the struct defs down so they're after the comment.